contracts/OrcaProtocol.sol

```diff
@@ -42,21 +46,85 @@ contract OrcaProtocol {
         uint256 _podId,
         uint256 _totalSupply,
         uint256 _votingPeriod,
-        uint256 _minQuorum,
-        address _gnosisMasterContract
+        uint256 _minQuorum
     ) public {
         // add a require to confirm minting was successful otherwise revert
         powerBank.createPod(msg.sender, _podId, _totalSupply);
 
-        voteManager.setupPodVotingAndSafe(
+        voteManager.createVotingStrategy(
             _podId,
             _votingPeriod,
-            _minQuorum,
-            _gnosisMasterContract
+            _minQuorum
         );
+        address podSafe = safeTeller.createSafe(_podId);
+
+        safeAddress[_podId] = podSafe;
+
         emit CreatePod(_podId);
     }
 
+    function createRuleProposal(
+        uint256 _podId,
+        address _contractAddress,
+        bytes4 _functionSignature,
+        bytes32[5] memory _functionParams,
+        uint256 _comparisonLogic,
+        uint256 _comparisonValue
+    )public{
+
+        require(powerBank.getPower(msg.sender, _podId) != 0 ,"User lacks power");
+
+        ruleManager.setPodRule(        
+        _podId,
+        _contractAddress,
+        _functionSignature,
+        _functionParams,
+        _comparisonLogic,
+        _comparisonValue);
+
+        voteManager.createProposal(
+            _podId,
+            msg.sender,
+            0,
+            _podId
+        );
+
+    }
+
+    function createActionProposal(uint256 _podId, address _to, uint256 _value, bytes memory _data)public{
```

Comment:
Do we have docs that explain what an action proposal is? I think RuleProposals are fairly self explanatory (though it'd be good to document that too), but action proposals are a little more ambiguous.